ui/lib/kv/addon/components/page/secret/patch.hbs

```diff
@@ -0,0 +1,64 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+<KvPageHeader @breadcrumbs={{@breadcrumbs}} @pageTitle="Patch Secret to New Version" />
+
+<MessageError @errorMessage={{this.errorMessage}} />
+
+<div class="box is-sideless is-fullwidth is-bottomless">
+  <NamespaceReminder @mode="patch" @noun="secret" />
+  <Hds::Form::TextInput::Field name="secret path" @value={{@path}} disabled data-test-field="Path" as |F|>
+    <F.Label>Path for this secret</F.Label>
+  </Hds::Form::TextInput::Field>
+
+  <hr class="has-background-gray-200" />
+
+  <Hds::Text::Display @tag="h2" @size="300" class="has-bottom-padding-s">Patch secret data</Hds::Text::Display>
+  <Hds::Alert @type="compact" @icon="alert-triangle-fill" @color="warning" class="has-bottom-padding-m" as |A|>
+    <A.Description>
+      The
+      <code>PATCH</code>
+      action allows you to partially update or add a key-value pair to the current version of the secret. The values will
+      remain the same in the new version if no changes are made to them.
+    </A.Description>
+  </Hds::Alert>
+
+  <Hds::Form::Radio::Group @name="patch-method" class="has-bottom-margin-m" as |G|>
+    <G.Legend>Edit via</G.Legend>
+    <G.HelperText>
+      Choose how to patch the secret data.
+      <strong>Switching to another method will reset the form data.</strong>
+    </G.HelperText>
+
+    {{#each (array "JSON" "UI") as |method|}}
+      <G.RadioField
+        @value={{method}}
+        {{on "change" this.selectPatchMethod}}
+        checked={{eq this.patchMethod method}}
+        data-test-input={{method}}
+        as |F|
+      >
+        <F.Label>{{method}}</F.Label>
+      </G.RadioField>
+    {{/each}}
+  </Hds::Form::Radio::Group>
+
+  {{#if (eq this.patchMethod "UI")}}
+    <KvPatch::Editor::Form
+      @isSaving={{this.save.isRunning}}
+      @onCancel={{this.onCancel}}
+      @onSubmit={{perform this.save}}
+      @subkeys={{@subkeys}}
```

Comment:
I feel like being able to look at the subkeys would also be useful for the JSON editor -- could we move that out of this form so both could use it?

Reply:
Yes! Great idea, I was wondering about this as well.

Reply:
Because the submit/cancel buttons are rendered in the child components, I can't pull it out and render it in a single place in the parent. 
A few options:
- yield it in both forms from the patch parent component (patch component then doesn't pass `@subkeys` arg and handles the toggle action)
- both forms render it separately
- move it another place (below the buttons? or above the form input, but I think that could be confusing...)

leaning towards both forms rendering it separately but let me know if you have another thought 😄

Reply:
I slightly prefer yielding so that we don't need to update it 2 places, but any of these seem reasonable!

Reply:
It'd be duplicate code in the parent component, which looked a little odd to me but it's nice that the subkey toggle action is then managed in just the parent